File: src/Python/npc_generator.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
import pathlib
import os
import shutil

def generateNpc():
	
  npc_name = input("What is the name of your npc that you create?\n")
  npc_count = int(input("How much npc you created in the map?\n"))
  main_script = input("Enter the main spawn script:\n")
  counter = 0
  lua_command = f'dofile(GetScriptPath().."{main_script}.lua")'
  spawn_template = '''State
{
	StateName = "INIT",
	OnFigureRespawnEvent
	{
		WaitTime = 10,
		X = 119,
		Y = 103,
		Conditions =
		{
		},
		Actions =
		{
		},
	},
}
'''
  
  f = open(f'{main_script}.lua','w')
  f.write(spawn_template)
  f.close()
  
  for counter in range(npc_count):
    f = open(f'{npc_name}{counter + 1}.lua','w')
    f.write(lua_command)
    f.close()
    
generateNpc()

def generateNpcActions():
  spellForce2_actions = input("Enter the lua function you need for your npc's ")
  npc_count = int(input("How much npc you created in the map?\n"))
  tag_npc_name = input("Enter your currect npc name tag from the map ")
  x_value = input("Insert the X value ")
  y_value = input("Insert the X value ")
  counter = 0
  
  for counter in range(npc_count):
    print(spellForce2_actions + '{Tag="'+tag_npc_name + f'{counter + 1}' + '",X=' + x_value + ',Y='+y_value + '},')
generateNpcActions()

File: src/JavaScript/npc_generator.js
#!/usr/bin/env node

const fs = require('fs');
const readline = require('readline');

function generateNpc(){
	
	let userInput = readline.createInterface({
		
		input:process.stdin,
		output:process.stdin
		
	});
	
	userInput.question('What is the name of your npc that you create in editor? -> ',npcName=>{
		
		userInput.question('How much npc your created in the map? -> ',npc_count =>{
			
	    userInput.question('Enter the main script filename that will be in the npc file you generate: -> ', mainScriptName => {
				
				let lua_command = `dofile(GetScriptPath().."${mainScriptName}.lua")`;
				let stateNpcTemplate = `State
{
	StateName = "INIT",
	OnFigureRespawnEvent
	{
		WaitTime = 10,
		X = 119,
		Y = 103,
		Conditions =
		{
		},
		Actions =
		{
		},
	},
}`;     let counter = 1;
        
        fs.writeFile(`${mainScriptName}.lua`,stateNpcTemplate,err=>{
					
					if(err){
						
						console.error(err);
						
					}
					
				});
        for(counter; counter <= npc_count; counter++){
					
					fs.writeFile(`${npcName}${counter}.lua`,lua_command,err=>{
						
						if(err){
							
						  console.error(err);
							
						}
						
					});
					
				}
		    userInput.close();
				generateNpcActions();
		
	    });
		
	  });
		
	});
	
}
generateNpc();

function generateNpcActions(){
	
	let userInput = readline.createInterface({
		
		input:process.stdin,
		output:process.stdin
		
	});
	
	userInput.question('Enter the lua function you need for your npcs -> ', npcAction=>{
		
	  userInput.question('How much npc your created in the map? -> ', npc_count=>{
			
		  userInput.question('Enter your currect npc name tag from the map -> ', tag_name=>{
				
			  userInput.question('Insert the X value -> ',xValue=>{
					
				  userInput.question('Insert the Y value -> ',yValue=>{
						
						let counter = 1;
						
						for(counter; counter <= npc_count; counter++){
							
							console.log(`${npcAction}{Tag="${tag_name}${counter}", X = ${xValue}, Y = ${yValue}},`);
							userInput.close();
						}
						
					});
					
				});	
				
			});	
			
		});
		
	});
	
}